### PR TITLE
Split parquet bloom filter config and enable bloom filter on read by default

### PR DESCRIPTION
**Pull request by lewiszlw**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion/issues/10299.

### CODE REVIEW

File: datafusion/sqllogictest/test_files/predicates.slt [hunk 1]
```diff
@@ -515,7 +515,7 @@ statement ok
 CREATE EXTERNAL TABLE data_index_bloom_encoding_stats STORED AS PARQUET LOCATION '../../parquet-testing/data/data_index_bloom_encoding_stats.parquet';
 
 statement ok
-set datafusion.execution.parquet.bloom_filter_enabled=true;
+set datafusion.execution.parquet.bloom_filter_on_read_enabled=true;
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/predicates.slt:518]: It seems like this test was to ensure we had end to end coverage, so now that we switched the default the meaning / coverage has changed

Perhaps we can change it so:
1. verify the setting with `show datafusion.execution.parquet.bloom_filter_on_read_enabled`
2. Add a second set of queries with `set datafusion.execution.parquet.bloom_filter_on_read_enabled=false`

**Review by alamb — COMMENTED** [review]
Thank you @lewiszlw  -- I think this looks great 🙏 

I had some suggestions on naming, comments, and tests. Let me know what you think

File: datafusion/core/src/datasource/physical_plan/parquet/mod.rs [hunk 2]
```diff
@@ -243,14 +243,32 @@ impl ParquetExec {
     }
 
     /// If enabled, the reader will read by the bloom filter
-    pub fn with_enable_bloom_filter(mut self, enable_bloom_filter: bool) -> Self {
-        self.table_parquet_options.global.bloom_filter_enabled = enable_bloom_filter;
+    pub fn with_enable_bloom_filter_on_read(
```
[the remark this answers is not in the document]
Comment (lewiszlw) [reply datafusion/core/src/datasource/physical_plan/parquet/mod.rs]: done

File: datafusion/common/src/config.rs [hunk 3]
```diff
@@ -395,8 +395,11 @@ config_namespace! {
         /// default parquet writer setting
         pub encoding: Option<String>, default = None
 
-        /// Sets if bloom filter is enabled for any column
-        pub bloom_filter_enabled: bool, default = false
+        /// Sets if bloom filter on read is enabled for any column
+        pub bloom_filter_on_read_enabled: bool, default = true
```
[the remark this answers is not in the document]
Comment (lewiszlw) [reply datafusion/common/src/config.rs]: Shorter name is bettter. Changed.

File: datafusion/sqllogictest/test_files/predicates.slt (hunk 1 above)
Comment (lewiszlw) [reply datafusion/sqllogictest/test_files/predicates.slt]: Done

File: datafusion/sqllogictest/test_files/predicates.slt [hunk 4]
```diff
@@ -514,8 +517,38 @@ DROP TABLE t;
...
+
+
+########
+# Test query without bloom filter
+# Refer to https://github.com/apache/datafusion/pull/7821#pullrequestreview-1688062599
+########
+
 statement ok
-set datafusion.execution.parquet.bloom_filter_enabled=true;
+set datafusion.execution.parquet.bloom_filter_on_read=false;
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/predicates.slt:551]: 👍

**Review by alamb — APPROVED** [review]
Awesome -- thank you @lewiszlw . This is a really nice improvement

cc @hiltontj

**Comment by hiltontj** [comment]
Thank you @lewiszlw for taking this one on!

**Comment by alamb** [comment]
Thanks again @lewiszlw

**Pull request merged**